test/Table_Tests/src/Database/SQLite_Spec.enso

```diff
@@ -252,53 +335,86 @@ sqlite_spec connection prefix =
-    sqlite_specific_spec prefix connection setup
-    Common_Table_Operations.Main.spec setup
+    agg_table_fn = _ ->
+        connection = create_connection_func Nothing
+        agg_in_memory_table.select_into_database_table connection (Name_Generator.random_name "Agg1") primary_key=Nothing temporary=True
+
+    empty_agg_table_fn = _ ->
+        connection = create_connection_func Nothing
+        (agg_in_memory_table.take (First 0)).select_into_database_table connection (Name_Generator.random_name "Agg_Empty") primary_key=Nothing temporary=True
+
+    setup = Common_Table_Operations.Main.Test_Setup.Config prefix agg_table_fn empty_agg_table_fn table_builder materialize is_database=True test_selection=common_selection aggregate_test_selection=aggregate_selection create_connection_func=create_connection_func
+    sqlite_specific_spec suite_builder prefix create_connection_func setup
+    Common_Table_Operations.Main.add_specs suite_builder setup
 
-    connection.close
 
 ## PRIVATE
 supported_replace_params : Set Replace_Params
 supported_replace_params =
     e = [Replace_Params.Value Text Case_Sensitivity.Default False, Replace_Params.Value Text Case_Sensitivity.Sensitive False, Replace_Params.Value Text Case_Sensitivity.Default True, Replace_Params.Value Text Case_Sensitivity.Sensitive True, Replace_Params.Value Text Case_Sensitivity.Insensitive True]
     Set.from_vector e
 
-spec =
-    enso_project.data.create_directory
-    file = enso_project.data / "transient" / "sqlite_test.db"
-    file.delete_if_exists
+backing_file =
+    transient_dir = enso_project.data / "transient"
+    assert transient_dir.exists ("There should be .gitignore file in the " + transient_dir.path + " directory")
+    transient_dir / "sqlite_test.db"
+
+
+create_inmem_connection =
+    Database.connect (SQLite In_Memory)
+
+
+create_file_connection file =
+    connection = Database.connect (SQLite file)
+    connection.execute_update 'CREATE TABLE "Dummy" ("strs" VARCHAR, "ints" INTEGER, "bools" BOOLEAN, "reals" REAL)'
+    connection
+
+
+type File_Connection
+    Value ~file
+
+    setup = File_Connection.Value <|
+        tmp_file = backing_file
+        con = create_file_connection backing_file
+        con.close
+        assert tmp_file.exists
+        tmp_file
+
+
+    teardown self =
+        assert self.file.exists
+        self.file.delete
+
+
+add_specs suite_builder =
     in_file_prefix = "[SQLite File] "
-    sqlite_spec (Database.connect (SQLite file)) in_file_prefix
-    Transaction_Spec.spec (Database.connect (SQLite file)) in_file_prefix
-    Upload_Spec.spec (_ -> Database.connect (SQLite file)) in_file_prefix
-    file.delete
+
+    sqlite_spec suite_builder in_file_prefix (_ -> create_file_connection backing_file)
+    Transaction_Spec.add_specs suite_builder (_ -> create_file_connection backing_file) in_file_prefix
+    Upload_Spec.add_specs suite_builder (_ -> create_file_connection backing_file) in_file_prefix
```

Comment:
Now, connecting creates the 'Dummy' table everywhere but this table was only supposed to be there for the `SQLite_Format should allow connecting to SQLite files` tests.

Can we restructure it to only create this table in that case?